From 7e8ed9c0f426bf5aac758a59a31e0d3ebcaef6c2 Mon Sep 17 00:00:00 2001
From: Kevin Mihelich <kevin@archlinuxarm.org>
Date: Mon, 1 Feb 2021 09:40:42 -0700
Subject: [PATCH] remove duplicate ProcessorCache declaration

Fixes multiple declaration error on ARM:
/usr/bin/ld: src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Processor.cpp.o:(.rodata+0x0): multiple definition of `typeinfo name for OpenColorIO_v2_0::ProcessorCache<unsigned int, std::shared_ptr<OpenColorIO_v2_0::Processor> >'; src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Config.cpp.o:(.rodata+0x1d4): first defined here
/usr/bin/ld: src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Processor.cpp.o:(.data.rel.ro+0x0): multiple definition of `typeinfo for OpenColorIO_v2_0::ProcessorCache<unsigned int, std::shared_ptr<OpenColorIO_v2_0::Processor> >'; src/OpenColorIO/CMakeFiles/OpenColorIO.dir/Config.cpp.o:(.data.rel.ro+0x0): first defined here
---
 src/OpenColorIO/Config.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/OpenColorIO/Config.cpp b/src/OpenColorIO/Config.cpp
index a207e439..07b2c9f2 100644
--- a/src/OpenColorIO/Config.cpp
+++ b/src/OpenColorIO/Config.cpp
@@ -1140,10 +1140,6 @@ public:
 
 
 
-// Instantiate the cache with the right types.
-template class ProcessorCache<std::size_t, ProcessorRcPtr>;
-
-
 ///////////////////////////////////////////////////////////////////////////
 
 ConfigRcPtr Config::Create()
-- 
2.28.0

